numba/cuda/cudadrv/driver.py

```diff
@@ -2605,15 +2609,27 @@ def error_log(self):
     def add_ptx(self, ptx, name):
         """Add PTX source in a string to the link"""
 
+    @abstractmethod
+    def add_cu(self, cu, name):
```

Comment:
Is the `name` here e.g. the file name from which the `cu` string was obtained?

Reply:
It could be, but it's not required.

Reply:
Should the docstring explain what it is? If it's not required should there be a default?

Reply:
Sorry, I mean it's not required to be the filename from which the cu string was obtained. However, it should be provided and should end in .cu - I've updated the docstring.

Reply:
Since I'm now using splitext (as you suggested OOB) it doesn't really matter what the extension is, so I've removed that requirement from the docstring.